src/super_gradients/common/plugins/deci_client.py

```diff
@@ -66,33 +71,32 @@ def _get_file(self, model_name: str, file_name: str) -> str:
                 logger.error(f"Deci client: {json.loads(e.body)['message']}")
             else:
                 logger.debug(e.body)
-
             return None
-        return FilesDataInterface.download_temporary_file(file_url=download_link)
 
-    def _get_model_cfg(self, model_name: str, cfg_file_name: str) -> DictConfig:
-        if not client_enabled:
-            return None
+        return FilesDataInterface.download_temporary_file(file_url=download_link)
 
-        file = self._get_file(model_name=model_name, file_name=cfg_file_name)
-        if file is None:
+    def get_model_arch_params(self, model_name: str) -> Optional[DictConfig]:
+        """Get the model arch_params from DeciPlatform.
+        :param model_name:  Name of the model as saved in the platform.
+        :return:            arch_params. None if arch_params were not found for this specific model on this SG version."""
+        arch_params_file = self._get_file(model_name, AutoNACFileName.STRUCTURE_YAML)
+        if arch_params_file is None:
             return None
-
-        split_file = file.split("/")
-        with hydra.initialize_config_dir(config_dir=normalize_path(f"{'/'.join(split_file[:-1])}/"), version_base=None):
-            cfg = hydra.compose(config_name=split_file[-1])
-        return cfg
-
-    def get_model_arch_params(self, model_name: str) -> DictConfig:
-        return self._get_model_cfg(model_name, AutoNACFileName.STRUCTURE_YAML)
-
-    def get_model_recipe(self, model_name: str) -> DictConfig:
-        return self._get_model_cfg(model_name, AutoNACFileName.RECIPE_YAML)
-
-    def get_model_weights(self, model_name: str) -> str:
-        if not client_enabled:
+        return _load_cfg(config_path=arch_params_file)
+
+    def get_model_recipe(self, model_name: str) -> Optional[DictConfig]:
+        """Get the model recipe from DeciPlatform.
+        :param model_name:  Name of the model as saved in the platform.
+        :return:            recipe. None if recipe were not found for this specific model on this SG version."""
+        recipe_file = self._get_file(model_name, AutoNACFileName.RECIPE_YAML)
+        if recipe_file is None:
             return None
+        return _load_cfg(config_path=recipe_file)
```

Comment:
I think that the code is easier to understand and more modular by splitting `_get_model_cfg` into `_get_file` and `_load_cfg`